addons/microsoft_calendar/models/calendar.py

```diff
@@ -74,13 +74,47 @@ def create(self, vals_list):
         if self._check_microsoft_sync_status() and not notify_context and recurrency_in_batch:
             self._forbid_recurrence_creation()
 
-        # for a recurrent event, we do not create events separately but we directly
-        # create the recurrency from the corresponding calendar.recurrence.
-        # That's why, events from a recurrency have their `need_sync_m` attribute set to False.
-        return super(Meeting, self.with_context(dont_notify=notify_context)).create([
-            dict(vals, need_sync_m=False) if vals.get('recurrence_id') or vals.get('recurrency') else vals
-            for vals in vals_list
-        ])
+        # Create events separately since the sender user may differ between events.
+        events = self.env['calendar.event']
+        users_by_id = self._get_users_by_id(vals_list)
+        vals_by_user = {}
```

Comment:
Use default dict!
```suggestion
        vals_by_user = defaultdict(lambda: [])
```

now you can do simply:

```py
for vals in vals_list:
    ...
    vals_by_user[sender_user].append(vals)
```